### PR TITLE
Updated the nelson ODE function, (made it nicer to look at) and added parameters

### PR DESCRIPTION
**Pull request by nnd389**
## Checklist

- [x] Appropriate tests were added
- [x] Any code changes were done in a way that does not break public API
- [x] All documentation related to code changes were updated
- [x] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [x] Any new documentation only uses public API
  
## Additional context

Here are the Nelson ODE updates! Feel free to take or leave any portion of these changes. I didn't find anywhere to add figures, but the plot that this code produces just shows a funky comparison of different algorithms. When I was searching for accurate, quick algorithms to use, I discovered that lsoda produces accurate results using the saveat parameter, but I'm still unsure as to why. 
Thanks!

### CODE REVIEW

File: benchmarks/AstroChem/nelson.jmd [hunk 1]
```diff
@@ -10,145 +10,176 @@ using Sundials, LSODA
...
+      0.0002,   # 12: C+
+      2.0e-7,   # 13: M+
+      2.0e-7]   # 14: M
+
+prob = ODEProblem(Nelson!, u0, tspan, params)
 refsol = solve(prob, Vern9(), abstol=1e-14, reltol=1e-14)
+sol1 = solve(prob, Rodas5P())
+sol2 = solve(prob, FBDF())
+sol3 = solve(prob, lsoda())
+sol4 = solve(prob, lsoda(), saveat = 1e10)
```
Comment (ChrisRackauckas) [inline benchmarks/AstroChem/nelson.jmd:170]: `saveat` does not change the accuracy of the solution. I'm curious, what are you trying to show with this part?

**Comment by nnd389** [comment]
When I graph lsoda by itself it produces an incorrect graph, but when I
graph it with the saveat parameter it produces a more accurate looking
graph. I know that it shouldn’t change the time step, but do you know why
lsoda produces a more accurate graphed result with saveat?

DESPOTIC, (the python code that I was comparing my results with) uses
scipy.integrate.odeint which uses lsoda from the FORTRAN library odepack.
And I couldn’t figure out how to get Julia lsoda to work until I used
saveat.

**Comment by ChrisRackauckas** [comment]
Thanks for the contributions! It needed a little bit more in the bump, which I incorporated into https://github.com/SciML/SciMLBenchmarks.jl/pull/1110. The built website should have your changes in about 2 days.

**Comment by nnd389** [comment]
Awesome! Thank you so much!

**Pull request closed without merging**